File: ui/src/model/monthlySummary.ts
export interface MonthlySummary {
  month: string;
  averageMonthlyRating: number;
  headChange: number;
  recordCount: number;
  finalReadings: {
    head: number;
    weight: number;
    price: number;
  }
}

export interface Summaries {
  summaries: MonthlySummary[];
}

File: src/main/java/com/atlascarbon/stockapi/dto/MonthlySummary.java
package com.atlascarbon.stockapi.dto;


import com.fasterxml.jackson.annotation.JsonFormat;

import java.time.YearMonth;

public record MonthlySummary(@JsonFormat(pattern = "MMM-yyyy", locale = "en_US") YearMonth month,
                             double averageMonthlyRating,
                             int headChange, int recordCount, FinalReadings finalReadings) {
}